samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartTestKit.java

```diff
@@ -0,0 +1,62 @@
+package com.example.shoppingcart;
+
+import com.example.shoppingcart.domain.ShoppingCartDomain;
+import com.example.shoppingcart.ShoppingCartApi;
+import com.google.protobuf.Empty;
+import java.util.ArrayList;
+import java.util.List;
+
+// Q Would we need ShoppingCartImpl as parameter constructor? 
+public class ShoppingCartTestKit {
+
+    ShoppingCartDomain.Cart currentState = ShoppingCartDomain.Cart.newBuilder().build();
+
```

Comment:
This is the question that concerns me the most by now

Reply:
Yeah, we can't know, so that will have to be part of the assertions possible on the result, one suggestion is something like the Akka testkit probes:

```
MyEvent event = result.expectEventOfType(MyEvent.class);
assertEquals(event.something, "some value");
```

This is easy for the normal case of one event, but for multiple it is a bit trickier to provide something convenient, but perhaps `result.expectNthEventOfType(index, MyEvent.class)` or something along those lines?

Reply:
taking inspiration from the actor testkit, perhaps invoking `expectEventOfType()` multiple times should return the subsequent events?

Reply:
`com.google.protobuf.GeneratedMessageV3` seems the supertype

Reply:
Good point guys, `expectEventOfType` looks good. I think calling `expectEventOfType()` multiple times is also in tune with the `def event` in `EventSourcedBehaviorTestKitImpl`.

I should create a similar `def event` as well now that I think about it.